src/probnum/filtsmooth/optim/_state_space_optimizer.py

```diff
@@ -0,0 +1,32 @@
+"""Gauss-Newton methods in state-space models."""
+
+import abc
+
+from probnum.filtsmooth.optim import _stoppingcriterion
+
+
+class StateSpaceOptimizer(abc.ABC):
```

Comment:
> Do you think the name optim for the package could suggest that we optimize - for instance - hyperparameters for state-space model regression, or sth like that? Is it clear enough that the content of the package is itself an optimizer, rather than it does optimize sth.? (minor remark)

Same concern here

Reply:
Maybe this problem is resolved if I mention in the first docstring line that the optimiser -- in the current concept --  computes states, not parameters?

Reply:
Yep, that would help, thanks! :)